cpp/src/arrow/record_batch_test.cc

```diff
@@ -756,7 +763,8 @@ TEST_F(TestRecordBatch, ToTensorSupportedNullToNan) {
 
   auto batch = RecordBatch::Make(schema, length, {a0, a1});
 
-  ASSERT_OK_AND_ASSIGN(auto tensor, batch->ToTensor(/*null_to_nan=*/true));
+  ASSERT_OK_AND_ASSIGN(auto tensor,
+                       batch->ToTensor(/*null_to_nan=*/true, /*row_major=*/false));
```

Comment:
We should probably also test the `null_to_nan` option for row major as well?

Would it be possible to test both inside this function? Only when creating the expected tensor, the values need to be passed in a different order?